src/layout-persistence/dotnet/src/MorganStanley.ComposeUI.LayoutPersistence/FileLayoutPersistence.cs

```diff
@@ -0,0 +1,101 @@
+﻿/*
+ * Morgan Stanley makes this available to you under the Apache License,
+ * Version 2.0 (the "License"). You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership. Unless required by applicable law or agreed
+ * to in writing, software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+
+using MorganStanley.ComposeUI.LayoutPersistence.Abstractions;
+
+namespace MorganStanley.ComposeUI.LayoutPersistence;
+
+public class FileLayoutPersistence : ILayoutPersistence
+{
+    private readonly string _basePath;
+    private readonly SemaphoreSlim _semaphore = new(1,1);
+
+    public FileLayoutPersistence(string basePath)
+    {
+        _basePath = NormalizeFilePath(basePath);
+
+        try
+        {
+            if (!Directory.Exists(_basePath))
+            {
+                Directory.CreateDirectory(_basePath);
+            }
+        }
+        catch (Exception ex)
+        {
+            throw new IOException($"Failed to create or access the directory: {_basePath}.", ex);
+        }
+    }
+
+    public async Task SaveLayoutAsync(string layoutData, string layoutName, CancellationToken cancellationToken = default)
+    {
+        var filePath = GetFilePath(layoutName);
+
+        await _semaphore.WaitAsync(cancellationToken);
+
+        try
+        {
+            await File.WriteAllTextAsync(filePath, layoutData, cancellationToken);
+        }
+        catch (Exception ex)
+        {
+            throw new IOException($"Failed to save layout to file: {filePath}.", ex);
+        }
+        finally 
+        {
+            _semaphore.Release();
+        }
+    }
+
+    public async Task<string> LoadLayoutAsync(string layoutName, CancellationToken cancellationToken = default)
+    {
+        var filePath = GetFilePath(layoutName);
+
+        await _semaphore.WaitAsync(cancellationToken);
+
+        if (!File.Exists(filePath))
+        {
+            throw new FileNotFoundException($"Layout file not found: {filePath}");
+        }
+
+        try
+        {
+            return await File.ReadAllTextAsync(filePath, cancellationToken);
+        }
+        catch (Exception ex)
+        {
+            throw new IOException($"Failed to load layout from file: {filePath}.", ex);
+        }
+        finally
+        { 
+            _semaphore.Release(); 
+        }
+    }
+
+    private string GetFilePath(string layoutName)
+    {
+        return Path.Combine(_basePath, $"{layoutName}.layout");
```

Comment:
While we only intend file persistence as a developer tool, I'd still care about path traversal here and restrict the layout file location to the _basePath folder